### DISCUSSION TITLE
LLDP Neighbors not showing

### DISCUSSION THREAD

**Discussion by redespace** [nautobot/nautobot#4441] [Q&A] [answered]
Hi Everyone,

I am not sure if I am missing something, but I am not able to see LLDP Neighbors. I am able to retrieve **Status** and **Configuration** successfully but not the LLDP Neighbors (After waiting a couple of seconds the table returns empty). The device has the _Secret Group_ related to it, with _Generic_ access type (Username, Password, Secret). Also, the platform has the correct driver mapping (napalm: ios).

### Environment
* Nautobot version (Docker tag too if applicable): V1.6.2
* Python version: 3.9.18 (napalm==4.1.0)
* Database platform, version: Postgres

I followed the steps described in the [(https://docs.nautobot.com/projects/core/en/stable/additional-features/napalm/](https://docs.nautobot.com/projects/core/en/stable/additional-features/napalm/). Also, the NAPALM support matrix should be able to get that info from a cisco ios: [https://napalm.readthedocs.io/en/latest/support/index.html#getters-support-matrix](https://napalm.readthedocs.io/en/latest/support/index.html#getters-support-matrix)
Is this a bug or I am doing something wrong?

**Comment by jdrew82** [comment]
This should just be doing a simple `get_lldp_neighbors` getter with NAPALM to the device. Assuming that it's an IOS device like you stated I'd confirm that the account you're using to authenticate with has permissions to perform the related command.

**Reply by redespace** [reply]
Thank you @jdrew82 for your reply. The user account has full rights to connect to the devices through ssh and throw any commands (even privilege/configure). The other functionality is working OK (status and configuration), so that tells me the permissions must me right. Is there any specific permission to do _get_lldp_neighbors_ ? Is this a permission set up in Cisco ISE? or is this a permission in Nautobot? where to do this:

`To make NAPALM requests via the Nautobot REST API, a Nautobot user must have assigned a permission granting the napalm_read action for the device object type.`

**Reply by jdrew82** [reply]
The permissions I'm speaking of would be enforced by ISE or whatever authentication mechanism you're using with the Device in question. If you're sure that the account being used in this case has the correct permissions I would test the getter with a simple python script to the Device. It's either the `get_lldp_neighbors()` or `get_lldp_neighbors_detail()` getters being used for that tab. If you're still having issues after that you could enable DEBUG logging and watch the Nautobot process for the communication between your device and the instance. It's possible you're running into a timeout situation.

**Reply by nkallergis** [reply]
As @jdrew82 suggests, I'd also start from the device's ISE/TACACS permissions. If this is a non-prod device, a quick alternative to writing a Python script would be to enable IOS CLI accounting to see what commands are actually executed.
```
event manager applet CLIaccounting
event cli pattern ".*" sync no skip no
action 1.0 syslog priority informational msg "$_cli_msg"
set 2.0 _exit_status 1
```
(stolen from Ivan Pepelnjak)

**Reply by redespace** [reply]
Thanks for the tip! I tried running the following python script, and is working well! It's returning all neighbor's information successfully with the same credentials. Nevertheless, in Nautobot is still not working :(


```
from getpass import getpass

from napalm import get_network_driver

driver = get_network_driver("ios")
device = driver(
    hostname="myhost",
    username="myuser",
    password=getpass("Pass: ")
)

device.open()
print(device.get_lldp_neighbors_detail())
device.close()

```

Any other ideas?

**Comment by nkallergis** [comment]
Ok, that pretty much rules out ISE/TACACS permissions which makes your issue even weirder. When clicking on the LLDP tab, do you see a connection attempt on the device?

**Reply by redespace** [reply]
Yes, I see the wheel rolling for like 5-6 secs, and then empty table:

![image](https://github.com/nautobot/nautobot/assets/40544192/a021eb09-7cc7-407b-92df-a69fdeaec32d)

But, if I click Status for example, then after 8-10 secs I get the data filled in correctly:

![image](https://github.com/nautobot/nautobot/assets/40544192/74bcbf26-604e-47fa-913a-085b351e860b)

**Reply by nkallergis** [reply]
Sorry I wasn't clear: is it possible to use the snippet I previously pasted on the actual device? That should give some insight on what Nautobot actually tries to do with NAPALM.

**Reply by redespace** [reply]
Ohh, in production environment we can't apply cli scripts on devices. Unfortunately I don't have a lab device to test on right now :(

**Reply by nkallergis** [reply]
Ah, I see. In this case, enabling Nautobot's DEBUG is the only idea I've got left but even this should be done with caution if we're talking about a production environment.

**Comment by travisjohnsonga** [comment]
I have the same issue - interfaces listed, but no LLDP info displayed in the web interface

**Reply by geordie12311** [reply]
Same here, get the status and configuration but no LLDP neighbors information

**Reply by redespace** [reply, marked as answer]
Actually I found out that for the LLDP Neighbors to show up, you first need to create ALL the interfaces of the device (or the ones you want to see neighbors on). Only then, it will show you what is the LLDP neighbor for each interface
